backend/src/names.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+/* Triton expects certain definitions within its backend libraries to follow
+ * specific naming conventions. Specifically, for a backend named
+ * "tools_identity," most definitions should appear within a namespace called
+ * triton::backend::tools_identity.
+ *
+ * In order to facilitate this with minimal effort on the part of backend
+ * developers, we ask that you put the name of your backend here. This macro is
+ * then used to propagate the correct namespace name wherever it is needed in
+ * the impl and interface code. */
+
+#define NAMESPACE tools_identity
```

Comment:
1. I notice we don't have a README for the backend folder. Can we add one? (Can be a separate ticket)
2. Assuming we have a README, this `NAMESPACE` concept seems like something we should document there for users.

Reply:
Let me take another look at the namespacing, I think it is different from the original design in rapids-triton noew because we replace it with `triton::developer_tools`, the name lookup is not direct and I did modify some places for proper looking (explicitly added `triton::backend::`)

Reply:
Let's hold on on this (and this PR generally). We have extensive docs for the backend wrappers, and we need to figure out where they will live in relation to this repo. Most importantly, we'll need to figure out where the backend template repo will live, since it needs to be a truly separate GH repo.

In terms of this specific issue, yes, we still need this `NAMESPACE` technique, but its use is documented elsewhere. This is for generating a _specific_ backend using the wrappers; it is not a namespace that appears generally in the wrappers themselves.